nostr/relay.py

```diff
@@ -54,42 +51,55 @@ def __post_init__(self):
             on_error=self._on_error,
             on_close=self._on_close
         )
+        self._connection_thread: Thread = None
 
-    def connect(self):
-        self.ws.run_forever(
-            sslopt=self.ssl_options,
-            http_proxy_host=self.proxy_config.host if self.proxy_config is not None else None, 
-            http_proxy_port=self.proxy_config.port if self.proxy_config is not None else None,
-            proxy_type=self.proxy_config.type if self.proxy_config is not None else None,
-        )
+    def connect(self, is_reconnect=False):
+        if not self.is_connected():
+            with self.lock:
+                self._connection_thread = Thread(
+                    target=self.ws.run_forever,
+                    kwargs={
+                        "sslopt": self.ssl_options,
+                        "http_proxy_host": self.proxy_config.host if self.proxy_config is not None else None,
+                        "http_proxy_port": self.proxy_config.port if self.proxy_config is not None else None,
+                        "proxy_type": self.proxy_config.type if self.proxy_config is not None else None
+                    },
+                    name=f"{self.url}-connection"
+                )
+                self._connection_thread.start()
+
+            if not is_reconnect:
```

Comment:
Would be more direct here to safe this as self._outgoing_message_thread, and check that directly.

```
__init__():
  ...
  self._outgoing_message_thread = None

def connect(self):
  ...
  if not self._outgoing_message_thread:
    self._outgoing_message_thread = Thread(...)

Reply:
Then no `is_reconnect` needed

Reply:
Couldn't its instantiation also just be moved into the `__post_init__`? Doesn't seem like there would ever a case where it would have to be re-created.

Reply:
It could be I don't think it makes sense to start the `outgoing_message_thread` until the connection is made.